server/src/main/java/org/apache/druid/client/indexing/ClientCompactionTaskQuery.java

```diff
@@ -43,14 +42,14 @@
 
   @JsonCreator
   public ClientCompactionTaskQuery(
-      @JsonProperty("id") @Nullable String id,
+      @JsonProperty("id") String id,
       @JsonProperty("dataSource") String dataSource,
       @JsonProperty("ioConfig") ClientCompactionIOConfig ioConfig,
       @JsonProperty("tuningConfig") ClientCompactionTaskQueryTuningConfig tuningConfig,
       @JsonProperty("context") Map<String, Object> context
   )
   {
-    this.id = id == null ? IdUtils.newTaskId(TYPE, dataSource, null) : id;
+    this.id = Preconditions.checkNotNull(id, "id");
```

Comment:
For backwards compatibility, would it be better to allow a null id as before?

Reply:
Seems like you were seeing changes between commits. The `id` field was newly added in this PR. Before this PR, `id` was initialized when `ClientCompactionTaskQuery` is converted into `CompactionTask`. Since the `id` field in `ClientCompactionTaskQuery` was never accessed on the coordinator side, I think there should be no backward-compatibility issue.